ext/src/ruby_api/wasi_config.rs

```diff
@@ -215,27 +226,35 @@ impl WasiCtxBuilder {
     /// Set the arguments (argv) to the specified +Array+.
     /// @param args [Array<String>]
     /// @def set_argv(args)
-    /// @return [WasiCtxBuilder] +self+
+    /// @return [WasiConfig] +self+
     pub fn set_argv(rb_self: RbSelf, argv: RArray) -> RbSelf {
         let mut inner = rb_self.inner.borrow_mut();
         inner.args = Some(argv.into());
         rb_self
     }
 
-    pub fn build(ruby: &Ruby, rb_self: RbSelf) -> Result<WasiCtx, Error> {
-        let mut builder = wasi_common::sync::WasiCtxBuilder::new();
-        let inner = rb_self.inner.borrow();
+    pub fn build(&self, ruby: &Ruby) -> Result<WasiP1Ctx, Error> {
+        let mut builder = WasiCtxBuilder::new();
+        let inner = self.inner.borrow();
 
         if let Some(stdin) = inner.stdin.as_ref() {
             match stdin {
                 ReadStream::Inherit => builder.inherit_stdin(),
                 ReadStream::Path(path) => {
-                    builder.stdin(file_r(ruby.get_inner(*path)).map(wasi_file)?)
+                    // Reading the whole file into memory and passing it as an
+                    // in-memory buffer because I cannot find a public struct
+                    // to use a file as an input that implements `StdinStream`
+                    // and the implementation would not be trivial.
```

Comment:
The feedback in [this Zulip conversation](https://bytecodealliance.zulipchat.com/#narrow/channel/217126-wasmtime/topic/Using.20a.20file.20as.20the.20stdin.20stream/near/522656785) is we should consider opening a PR upstream to add an API to use a file.